### PR TITLE
Move `nsubsystems` from QOBase

### PR DESCRIPTION
**Pull request by Abhishek-1Bhatt**

### CODE REVIEW

**Comment by Krastanov** [comment]
Basis and CompositeBasis are defined here, so they are all right.

Also bump the patch version number so we can release it promptly.

**Comment by Krastanov** [comment]
could you bump the patch version and add to the changelog?

File: src/embed_permute.jl [hunk 1]
```diff
@@ -84,3 +84,11 @@ function embed(basis_l::CompositeBasis, basis_r::CompositeBasis,
 end
 
 permutesystems(a::AbstractOperator, perm) = arithmetic_unary_error("Permutations of subsystems", a)
+
+nsubsystems(s::AbstractKet) = nsubsystems(basis(s))
+function nsubsystems(s::AbstractOperator)
+    s.basis_l == s.basis_r || throw(ArgumentError("`nsubsystem(::AbstractOperator)` is well defined only if the left and right bases are the same"))
```
Comment (Krastanov) [inline src/embed_permute.jl:90]: same comment as with ::AbstractKet. We can not rely on AbstractOperator to have .basis_l and .basis_r fields.

Comment (Krastanov) [reply src/embed_permute.jl:90]: I think a basis(op) call will raise an error if the left and right basis are not the same, so maybe just use it directly

Comment (Abhishek-1Bhatt) [reply src/embed_permute.jl:90]: should we stick to throwing the current error message with the basis(op) in a try catch or just let the basis(op) error show when the bases are unequal?

Comment (Krastanov) [reply src/embed_permute.jl:90]: no need to stick to the current message, it is fine to rely on whatever message basis throws. Just check that it actually works (you know, never trust me, I can always be wrong ;)

**Pull request merged**